pkg/solana/chainwriter/transform_registry.go

```diff
@@ -190,6 +168,128 @@ func fetchPoolLookupAccounts(ctx context.Context, client client.MultiClient, poo
 	return poolAccounts, nil
 }
 
+func calculateComputeUnitLimit(argsTransformed ccipsolana.SVMExecCallArgs) (uint32, error) {
+	cu, ok := argsTransformed.ExtraData.ExtraArgsDecoded["computeUnits"].(uint32)
+	if !ok {
+		return 0, fmt.Errorf("computeUnits not found in ExtraData")
+	}
+
+	computeUnits := StaticCuOverhead + cu
+
+	for _, execData := range argsTransformed.ExtraData.DestExecDataDecoded {
+		destGasAmount, ok := execData["destGasAmount"].(uint32)
+		if !ok {
+			return 0, fmt.Errorf("DestGasAmount not found in ExtraData")
+		}
+		computeUnits += destGasAmount
+	}
+	return computeUnits, nil
+}
+
+func resolveCommonTokenTransferAccounts(ctx context.Context, tokenAccountsRequired bool, client client.MultiClient, toAddress string, args any, tableMap map[string]map[string][]*solana.AccountMeta) (commonTokenTransferAccounts, error) {
+	// Return empty struct if token accounts are not required
+	if !tokenAccountsRequired {
+		return commonTokenTransferAccounts{}, nil
+	}
+	registryTables, exists := tableMap["PoolLookupTable"]
```

Comment:
maybe we should also have a check for the size of the table here (or later when it's indexed)